---
layout: post
title: "TIL #48 리액트 fetch"
data: '2021-01-17'
category: [TIL]
---

오늘은 실시간 강의로 react에서 hook과 data fatching을 배웠다. 인강으로 강의를 듣고 수업을 들어서 이해가 좀더 잘 됐던것 같다. 개념하고 실습을 번갈아 가면서 진행해서 이해도가 좀금더 좋아 지는 것 같다. 다만 실습을 할때 자주 실수 하는게 있다. 오늘도 다 아는 문제들이고 풀수 있는 문제들이었다. 근데 자주 사용하던 코드들을 사용할때 상황을 생각하지 않고 내가 사용하고 싶은대로 작성하는 버릇이 있는것 같다. 여기서 어떤 데이터가 나올지 모르는데 아 여기는 이게 나오고 그러면 이걸쓰면 돼겠지 하면 에러가 뜬다. data fetching할때 처음부터 console.log찍으면서 데이터를 받았으면 금방했을 텐데 엉뚱한 부분을 계속 수정하고 있었다. 또 map을 쓰는데 요즘 많이 사용하다보니 맞겠지란 생각에 맘대로 작성한 부분이 있었다. {} () 괄호차이 하나로 결국 제시간에 문제를 못풀었다. 분명히 다 맞았는데 어디가 잘못됐는지 한참을 고민한 것같다. 근데 괄호 하나로 인해 안됐다고 하니 허무했다. 코드를 작성할때는 안다고 너무 생각없이 하면 안돼는거 같다. 작은 실수하나가 결국 오류로 끝나고 마는 경우들이 있는것 같다. 예전에도 결국 못찾아서 처음부터 다시 쓴 적도 있었다. 실수를 최대한 하지 않도록 노력 해야할 것 같다.

오늘 강의 data fetching 하기 실습

```jsx
import { useState } from "react";

const delay = () => new Promise((resolve) => setTimeout(resolve, 2000));

export const Poketmon = () => {
  const [poketmons, setPoketmons] = useState([]);
  const [status, setStatus] = useState("idle");
  const handleClick = async () => {
    setStatus("loading");
    try {
      await delay();
      const response = await fetch(
        "https://pokeapi.co/api/v2/pokemon?limit=10&offset=0"
      );
      const data = await response.json();
      setPoketmons(data.results);
      setStatus("succeeded");
    } catch (error) {
      setStatus("failed");
    }
  };

  return (
    <div>
      <button onClick={handleClick}>
        {status === "loading" ? "요청중" : "요청하기"}
      </button>
      <ul>
        {poketmons.map((poketmon) => (
          <li key={poketmon.name}>
            {poketmon.name} / {poketmon.url}
          </li>
        ))}
      </ul>
    </div>
  );
};
```